src/importer.rs

```diff
@@ -159,7 +161,18 @@ impl<C: ObjCache> Importer<C> {
                 let models = match self.cache.get_by_checksum(checksum) {
                     Some(models) => models.clone(),
                     None => {
-                        let (tobj_models, _) = obj_buf_into_tobj(&mut file_contents.as_slice())?;
+                        let (mut tobj_models, _) =
+                            obj_buf_into_tobj(&mut file_contents.as_slice())?;
+                        let mut index = 0;
+
+                        while index != tobj_models.len() {
+                            if tobj_models[index].mesh.positions.is_empty() {
```

Comment:
unless you are removing orphans (which you may, I just haven't noticed), its not the `mesh.positions` but rather `mesh.indices` that may make the mesh empty.

Reply:
True. How do you propose we best solve this? We could change the mesh constructors to return `Option`?

Reply:
no need for an `Option` here, i think. just check if the `indices` contain anything, instead of `positions`.

i think it's ok when the mesh constructor panicks. we should feed the mesh constructors valid data. 
therefore i§d suggest validating the imported data first. should be fairly simple:

- the faces should reference an existing vertex (and normal, if present) index
- the faces should contain three non-repetetive indices
- the mesh has to have at least one valid face

we don't need to remove orphans during import, but we need to decide what should we do if any of the above mentioned tests fails. i suggest not trying to autofix but rather fail the import instead.

Reply:
I agree with failing the import. We shouldn't let the user think everything went well while some models are missing.

Reply:
> the faces should reference an existing vertex (and normal, if present) index

There's unfortunatelly a bug in `tobj` that crashes conversion if there's index out of bound. I filed a bug report for it.

Reply:
do we have a workaround meanwhile?

Reply:
Not anything that would be easy to implement on our side. One thing we can do later, if it isn't fixed upstream, is to fork (or just copy, it's 1 file) the library and fix it on our side.

Reply:
that would be much appreciated! we can't wait until somebody sometime fixes a bug

Reply:
Let's give that maintainer at least couple of days before we fix this ourselves. I don't have capacity for it right now.  Would that be possible?